server/src/main/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhase.java

```diff
@@ -128,7 +126,18 @@ protected SearchPhase getNextPhase(SearchPhaseResults<CanMatchResponse> results,
     }
 
     private static boolean shouldSortShards(MinAndMax<?>[] minAndMaxes) {
-        return Arrays.stream(minAndMaxes).anyMatch(Objects::nonNull);
+        Class<?> clazz = null;
+        for (MinAndMax<?> minAndMax : minAndMaxes) {
+            if (clazz == null) {
+                clazz = minAndMax == null ? null : minAndMax.getMin().getClass();
+            } else if (minAndMax != null && clazz != minAndMax.getMax().getClass()) {
```

Comment:
It's slightly confusing that we use `getMin` for the initial `getClass` and then `getMax`. Use `getMin` all the time maybe?

Reply:
++, I pushed https://github.com/elastic/elasticsearch/pull/53397/commits/ef3df35c1d9c57d31187e0c02c1b07c69b1250dc